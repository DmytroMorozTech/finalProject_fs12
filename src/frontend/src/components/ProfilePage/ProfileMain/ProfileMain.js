import React from 'react'
import Typography from '@material-ui/core/Typography'
import PhotoCameraIcon from '@material-ui/icons/PhotoCamera'
import CreateIcon from '@material-ui/icons/Create'
import style from './styles'
import { Hidden, Link } from '@material-ui/core'
import SharedButton from '../../../shared/SharedButton/SharedButton'
import SmallDot from '../../../shared/SmallDot/SmallDot'
import BusinessTwoToneIcon from '@material-ui/icons/BusinessTwoTone'
import toggleModalAction from '../../../redux/Modal/modalActions'
import { EDIT_INTRO, UPLOAD_AVATAR_IMG, UPLOAD_PROFILE_BACKGROUND_IMG } from '../../../redux/Modal/modalTypes'
import { useDispatch } from 'react-redux'
import clsx from 'clsx'
import Image from '../../../../src/shared/Image/Image'

function ProfileMain (props) {
  const { numberOfConnections = 45, isEditable } = props
  const profile = props.profile
  const classes = style()
  const dispatch = useDispatch()
  const preventDefault = (event) => event.preventDefault()

  // const defaultBgUrl = 'url(https://res.cloudinary.com/dan-insta-step/image/upload/v1630405373/linkedin/general/u4aqln7amyyfdj0tehqy.png)'
  // const bgChosenByUser = `url(${profile.profileBgUrl})`
  // const profileBgImage = profile.profileBgPublicId ? bgChosenByUser : defaultBgUrl

  return (
    <div className={classes.root} >
      <Image
        imageUrl={profile.profileBgPublicId}
        className={classes.profileBackgroundImg}
        type={'profileBgImg'}
        alt={'profile background'}
      />

      <div className={classes.header}>
        <div
          className={clsx(classes.photoIcon, !isEditable && classes.hidden)}
          onClick={() => dispatch(toggleModalAction({ modalType: UPLOAD_PROFILE_BACKGROUND_IMG }))}>
          <PhotoCameraIcon fontSize="inherit" color={'primary'}/>
        </div>
      </div>
      <div>
        <div className={classes.row}>
          <Image
            imageUrl={profile.avatarUrl}
            onClickHandler={() => dispatch(toggleModalAction({ modalType: UPLOAD_AVATAR_IMG }))}
            className={classes.bigAvatar}
            type={'profileAvatar'}
            alt={'user avatar'}
          />
          <div
            className={clsx(!isEditable && classes.hidden)}
            onClick={() => dispatch(toggleModalAction({ modalType: EDIT_INTRO, profile: props.profile }))}>
            <CreateIcon className={classes.editName}/>
          </div>
        </div>
        <div className={classes.info}>
          <div className={classes.leftPanel}>
            <Typography variant="h3">
              {profile.fullName}
            </Typography>
            <Typography variant="body1">
              {profile.headline}
            </Typography>
            <div className={classes.info}>
              <Typography variant="body1" color="secondary">
                {profile.city}, {profile.country}
              </Typography>
              <SmallDot/>
              <Link onClick={preventDefault}>
                <Typography variant="body1" color="primary" className={classes.bold}>
                  Contact info
                </Typography>
              </Link>
            </div>
            <div>
              <Link href="#" onClick={preventDefault}>
                <Typography variant="body1" color="primary" className={classes.bold}>
                  {numberOfConnections} connections
                </Typography>
              </Link>
            </div>
          </div>
          <div>
            <Hidden xsDown>
              <Typography variant="h5" className={classes.rightPanel}>
                <BusinessTwoToneIcon fontSize="large" color="secondary" className={classes.businessIcon}/>
                <span>{profile.positionAndCompany}</span>
              </Typography>
            </Hidden>
          </div>
        </div>
        <div>
          <SharedButton title="Open to"/>
          <SharedButton title="Add section" variant="outlined" color="secondary"/>
          <SharedButton title="More" variant="outlined" color="secondary"/>
        </div>
        <div className={classes.box}>
          <Link href="#" onClick={preventDefault} className={classes.column}>
            <Typography variant="body1" color="secondary" className={classes.bold}>Open to work</Typography>
            <Typography variant="body1" color="secondary">{profile.positionAndCompany}</Typography>
            <Typography variant="body1" color="primary" className={classes.bold}>See all details</Typography>
          </Link>
          <div className={clsx(!isEditable && classes.hidden)}>
            <CreateIcon fontSize="inherit" className={classes.createIcon}/>
          </div>
        </div>
      </div>
    </div>
  )
}

export default ProfileMain
