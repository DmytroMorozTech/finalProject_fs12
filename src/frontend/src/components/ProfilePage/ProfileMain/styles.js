import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  root: {
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    marginBottom: theme.spacing(2),
    padding: theme.spacing(4),
    backgroundSize: '100% 160px',
    backgroundRepeat: 'no-repeat'
  },

  header: {
    display: 'flex',
    justifyContent: 'flex-end'
  },

  photoIcon: {
    fontSize: theme.typography.icons.extraSmall.fontSize,
    width: theme.spacing(6),
    height: theme.spacing(6),
    padding: theme.spacing(1.5),
    backgroundColor: theme.palette.common.white,
    borderRadius: '50%',
    cursor: 'pointer'
  },

  row: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'space-between'
  },

  bigAvatar: {
    width: theme.spacing(31),
    height: theme.spacing(31),
    border: '4px solid white',
    borderRadius: '50%',
    cursor: 'pointer',
    marginBottom: theme.spacing(3)
  },

  editName: {
    cursor: 'pointer',
    color: theme.palette.grey[700],
    width: theme.spacing(7),
    height: theme.spacing(7),
    borderRadius: '50%',
    padding: theme.spacing(1.3),
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.1)'
    }
  },

  info: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: theme.spacing(1)
  },

  leftPanel: {
    minWidth: '68%'
  },

  rightPanel: {
    display: 'flex',
    alignItems: 'center'
  },

  businessIcon: {
    backgroundColor: theme.palette.grey[300],
    marginRight: theme.spacing(1)
  },

  bold: {
    fontWeight: theme.typography.fontWeightBold
  },

  box: {
    width: '50%',
    display: 'flex',
    justifyContent: 'space-between',
    backgroundColor: theme.palette.grey[200],
    borderRadius: theme.shape.small,
    marginTop: theme.spacing(2),
    padding: theme.spacing(2)
  },

  column: {
    display: 'flex',
    flexDirection: 'column'
  },

  createIcon: {
    cursor: 'pointer',
    color: theme.palette.grey[700],
    width: theme.spacing(3),
    height: theme.spacing(3)
  },
  hidden: {
    visibility: 'hidden'
  }
}))
