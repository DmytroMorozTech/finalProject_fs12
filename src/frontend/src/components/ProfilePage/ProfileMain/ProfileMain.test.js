import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import ProfileMain from './ProfileMain'

jest.mock('./styles', () => () => ({}))
jest.mock('react-router-dom', () => ({
  Link: 'a'
}))

const setUp = (props) => shallow(<ProfileMain {...props}/>)

describe('should render ProfileMain component', () => {
  let spyOnUseDispatch
  let mockDispatch

  const profile = {
    'id': 1,
    'createdDate': '2015-01-01T18:22:32.123123',
    'firstName': 'Richard',
    'lastName': 'West',
    'headline': 'Java Developer',
    'country': 'USA',
    'city': 'New York',
    'fullName': 'Richard West',
    'avatarPublicId': 'linkedin/avatars/vbalexhudya0ano14pgs',
    'profileBgPublicId': 'linkedin/profile-bg/isgzneqfm5ro1bufycqc',
    'positionAndCompany': 'Sales manager at Microsoft',
    'phoneNumber': '+380502926823',
    'age': '20',
    'email': 'test@gmail.com',
    'groups': [],
    'workPlaces': [
      {
        'id': 1,
        'position': 'Chief Operational Officer',
        'responsibilities': 'Quisque id rhoncus mauris. Phasellus quis diam aliquet, gravida massa in, suscipit felis. Aliquam ac turpis arcu. Nunc fermentum dui augue, et aliquam eros ullamcorper imperdiet.',
        'dateStart': '2000-12-31',
        'dateFinish': '2010-06-12',
        'isCurrentlyEmployed': false,
        'organization': {
          'id': 3,
          'name': 'PayPal Holdings Inc.',
          'location': '2211 N 1st St, San Jose, CA 95131, United States',
          'webSite': 'www.paypal.com'
        }
      },
      {
        'id': 2,
        'position': 'CEO',
        'responsibilities': 'Fusce lacinia, nisl et vestibulum ultricies, orci risus luctus urna, et consectetur sem nibh non mauris. Phasellus ac aliquam libero, eu luctus lorem. Nulla tempor risus eu tortor mattis, id eleifend tellus semper. ',
        'dateStart': '2010-06-20',
        'dateFinish': '2013-11-22',
        'isCurrentlyEmployed': false,
        'organization': {
          'id': 2,
          'name': 'Apple',
          'location': '1 Apple Park Way Cupertino, California, 95014-0642 United States',
          'webSite': 'apple.com'
        }
      },
      {
        'id': 3,
        'position': 'Sales manager',
        'responsibilities': 'Phasellus vitae interdum mauris, eget luctus neque. Aenean at venenatis mauris, venenatis convallis velit. Suspendisse hendrerit lectus dolor, sit amet tristique odio lobortis sit amet. Donec ante purus, ',
        'dateStart': '2014-01-20',
        'dateFinish': null,
        'isCurrentlyEmployed': true,
        'organization': {
          'id': 1,
          'name': 'Microsoft',
          'location': 'Redmond, WA, US (HQ) Microsoft Corporation One Microsoft Way',
          'webSite': 'microsoft.com'
        }
      }
    ],
    'educations': [
      {
        'id': 3,
        'school': 'MIT',
        'degreeReceived': 'Bachelor',
        'activities': 'Working hard; Writing scientifica articles',
        'description': 'High-level of education; nice capmus; ...',
        'fieldOfStudy': 'Electronics',
        'dateStart': '2011-03-03',
        'dateFinish': '2015-04-04'
      },
      {
        'id': 4,
        'school': 'Michigan state University',
        'degreeReceived': 'Msc',
        'activities': 'Doing mathematical calculations all day long',
        'description': 'Best calculus ever',
        'fieldOfStudy': 'Math',
        'dateStart': '2017-03-03',
        'dateFinish': '2019-03-03'
      }
    ],
    'certifications': [
      {
        'id': 1,
        'name': 'Spring data jpa',
        'issuingOrganization': 'DAN-IT',
        'hasExpiryDate': true,
        'issueDate': '2016-01-01',
        'expirationDate': '2026-01-01',
        'credentialId': 'PS35092',
        'credentialUrl': 'https://dan-it.com.ua'
      },
      {
        'id': 2,
        'name': 'On-site practice',
        'issuingOrganization': 'Unilever',
        'hasExpiryDate': false,
        'issueDate': '2017-07-08',
        'expirationDate': null,
        'credentialId': 'RQ3627AS',
        'credentialUrl': 'https://unilever.ua'
      }
    ],
    'isFollowedByActiveUser': false,
    'numberOfFollowers': 3
  }

  let isEditable = true

  let component

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp({profile, isEditable})
  })
  it('should work dispatch toggle modal upload profile bg', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModalUploadProfileBg\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
  it('should work dispatch upload avatar', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('Image[id=\'toggleModalUploadAvatar\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
  it('should work dispatch edit intro', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModalEditIntro\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
