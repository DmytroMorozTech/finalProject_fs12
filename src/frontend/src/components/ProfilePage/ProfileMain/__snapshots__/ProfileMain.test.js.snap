// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileMain component is editable with country, city, positionAndCompany to match snapshot 1`] = `
<div>
  <Image
    alt="profile background"
    imageUrl="linkedin/profile-bg/isgzneqfm5ro1bufycqc"
    type="profileBgImg"
  />
  <div>
    <div
      className=""
      id="toggleModalUploadProfileBg"
      onClick={[Function]}
    >
      <Memo(ForwardRef(PhotoCameraIcon))
        color="primary"
        fontSize="inherit"
      />
    </div>
  </div>
  <div>
    <div>
      <div
        id="toggleModalUploadAvatar"
        onClick={[Function]}
      >
        <Image
          alt="user avatar"
          className=""
          imageUrl="linkedin/avatars/vbalexhudya0ano14pgs"
          type="profileAvatar"
        />
      </div>
      <div
        className=""
        id="toggleModalEditIntro"
        onClick={[Function]}
      >
        <Memo(ForwardRef(CreateIcon)) />
      </div>
    </div>
    <div>
      <div>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Richard West
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          Java Developer
        </WithStyles(ForwardRef(Typography))>
        <div>
          <WithStyles(ForwardRef(Typography))
            color="secondary"
            id="cityAndCountry"
            variant="body1"
          >
            New York
            , 
            USA
          </WithStyles(ForwardRef(Typography))>
          <SmallDot />
          <WithStyles(ForwardRef(Link))
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(Typography))
              color="primary"
              variant="body1"
            >
              Contact info
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Link))>
        </div>
        <div
          id="numberOfConnections"
        >
          <WithStyles(ForwardRef(Link))
            href="#"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(Typography))
              color="primary"
              variant="body1"
            >
              55
               connections
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Link))>
        </div>
      </div>
      <div>
        <Hidden
          xsDown={true}
        >
          <WithStyles(ForwardRef(Typography))
            variant="h5"
          >
            <Memo(ForwardRef(BusinessTwoToneIcon))
              color="secondary"
              fontSize="large"
            />
            <span
              id="positionAndCompany"
            >
              Sales manager at Microsoft
            </span>
          </WithStyles(ForwardRef(Typography))>
        </Hidden>
      </div>
    </div>
    <div>
      <SharedButton
        title="Open to"
      />
      <SharedButton
        color="secondary"
        title="Add section"
        variant="outlined"
      />
      <SharedButton
        color="secondary"
        title="More"
        variant="outlined"
      />
    </div>
    <div>
      <WithStyles(ForwardRef(Link))
        href="#"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Typography))
          color="secondary"
          variant="body1"
        >
          Open to work
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="secondary"
          variant="body1"
        >
          Sales manager at Microsoft
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="primary"
          variant="body1"
        >
          See all details
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Link))>
      <div
        className=""
      >
        <Memo(ForwardRef(CreateIcon))
          fontSize="inherit"
        />
      </div>
    </div>
  </div>
</div>
`;
